diff --git a/binding.gyp b/binding.gyp
index bb40aa7e8d41644ba690761e9064fc1d0f2dcd6b..3abcecbd4966a00dfbe9b4d798ccd17b5290561c 100644
--- a/binding.gyp
+++ b/binding.gyp
@@ -24,14 +24,11 @@
       "defines": [
         "NAPI_VERSION=<(napi_build_version)",
       ],
-      "cflags_cc": [
-        "-std=c++17"
-      ],
       "conditions": [
         ["OS=='mac'", {
           "xcode_settings": {
             "GCC_SYMBOLS_PRIVATE_EXTERN": "YES", # -fvisibility=hidden
-            "CLANG_CXX_LANGUAGE_STANDARD": "c++17",
+            "CLANG_CXX_LANGUAGE_STANDARD": "<(cxxstd)",
             "MACOSX_DEPLOYMENT_TARGET": "10.9",
           },
         }],
@@ -39,14 +36,15 @@
           "msvs_settings": {
             "VCCLCompilerTool": {
               "AdditionalOptions": [
-                "/std:c++17",
+                "/std:<(cxxstd)",
               ],
               "RuntimeLibrary": 0,
             },
           },
         }],
-        ["OS == 'linux'", {
+        ["OS=='linux'", {
           "cflags_cc": [
+            "-std=<(cxxstd)",
             "-Wno-cast-function-type"
           ]
         }],
@@ -62,12 +60,21 @@
         "vendor/tree-sitter/lib/src",
         "vendor/tree-sitter/lib/include",
       ],
-      "cflags": [
-        "-std=c11"
-      ],
-      "defines": [
-        "_POSIX_C_SOURCE=200112L",
-        "_DEFAULT_SOURCE",
+      "conditions": [
+        ["OS=='win'", {
+          "defines": [
+            "_POSIX_C_SOURCE=200112L",
+            "_DEFAULT_SOURCE",
+          ],
+        }, {
+          "cflags": [
+            "-std=c11"
+          ],
+          "defines": [
+            "_POSIX_C_SOURCE=200112L",
+            "_DEFAULT_SOURCE",
+          ],
+        }],
       ]
     }
   ],
@@ -76,5 +83,6 @@
     "openssl_fips": "",
     "v8_enable_pointer_compression%": 0,
     "v8_enable_31bit_smis_on_64bit_arch%": 0,
+    "cxxstd%": "<!(node -p \"parseInt(process.env.npm_config_target ?? process.versions.node) < 22 ? 'c++17' : 'c++20'\")",
   }
 }
diff --git a/index.js b/index.js
index 9a57b3652ff41b8046c89c498802f6318ee78ff1..4c3141d291d3fcf51cf1370c4e0d64b57704eef4 100644
--- a/index.js
+++ b/index.js
@@ -1,7 +1,4 @@
-const binding = typeof process.versions.bun === "string" ?
-    // Statically analyzable enough for `bun build --compile` to embed the tree-sitter.node napi addon
-    require(`./prebuilds/${process.platform}-${process.arch}/tree-sitter.node`) :
-    require('node-gyp-build')(__dirname);
+const binding = require('node-gyp-build')(__dirname);
 const {Query, Parser, NodeMethods, Tree, TreeCursor, LookaheadIterator} = binding;
 
 const util = require('util');
